causallearn/utils/KCI/KCI.py

```diff
@@ -391,6 +401,9 @@ def kernel_matrix(self, data_x, data_y, data_z):
                     elif self.est_width == 'empirical':
                         kernelZ.set_width_empirical_kci(data_z)
                 Kzx = kernelZ.kernel(data_z)
+                Kzx = Kernel.center_kernel_matrix(Kzx)
+                # centering kernel matrix to conform with the original Matlab implementation,
+                # specifically, Line 100 in the file 'algorithms/CInd_test_new_withGP.m'
```

Comment:
But I only add the centering Kzx here (`self.kernelZ == 'Gaussian'`) , maybe I should also add `Kzx = Kernel.center_kernel_matrix(Kzx)` for `self.kernelZ == 'Polynomial'` and  `self.kernelZ == 'Linear'` ?

Reply:
hmmm, don't know. Probably need to ask KCI expert (who really knows a lot about KCI lol haha). Maybe ask in group chat? Maybe Mingming / Biwei knows?

Reply:
Thanks! Yeah I'm not sure.

`Kx` and `Ky` are centered by `center_kernel_matrix`, and then fed into `KCI_V_statistic`.

But in `KCI_V_statistic`, `Kzx` and `Kzy` are first centered by regression:

```python
KxR, Rzx = Kernel.center_kernel_matrix_regression(Kx, Kzx, self.epsilon_x)
```

Then, do we need to center them before `KCI_V_statistic`? Maybe we should ask experts.

If yes, except for the comments, we should also keep an eye on `Kzx` and `Kzy` - whether they are still the same, based on which we saved one computation before.

Reply:
> do we need to center them before `KCI_V_statistic`?
Matlab says yes... but I should check with the expert.

> If yes, except for the comments, we should also keep an eye on Kzx and Kzy - whether they are still the same, based on which we saved one computation before.
If it's a Gaussian kernel, then yes,  they are still the same. But I'm not sure about other kernels. I'll check with the expert.